src/main/kotlin/org/elm/ide/intentions/RecordConstructorToLiteralIntention.kt

```diff
@@ -0,0 +1,90 @@
+package org.elm.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.elm.lang.core.psi.ElmPsiFactory
+import org.elm.lang.core.psi.ElmTypes.UPPER_CASE_IDENTIFIER
+import org.elm.lang.core.psi.elements.*
+
+/**
+ * An intention action that transforms a usage of record constructor to the corresponding record literal, i.e. converts
+ * this:
+ * ```
+ * type alias MyTypeAlias = { foo: String, bar: Int }
+ *
+ * baz = MyTypeAlias "a" 1
+ * ```
+ * to this:
+ * ```
+ * type alias MyTypeAlias = { foo: String, bar: Int }
+ *
+ * baz = { foo = "a", bar = 1 }
+ * ```
+ */
+class RecordConstructorToLiteralIntention :
+    ElmAtCaretIntentionActionBase<RecordConstructorToLiteralIntention.Context>() {
+
+    /**
+     * @param functionCall The current record constructor function call, to be replaced with a record literal.
+     * @param arguments The arguments to the record literal. Each entry in the list is a pair of strings: the first is
+     * the argument name, the second is the argument's text.
+     */
+    data class Context(val functionCall: ElmFunctionCallExpr, val arguments: List<Pair<String, String>>)
+
+    override fun getText() = "Use record literal"
+    override fun getFamilyName() = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        // This intention is valid in a situation like this:
+        //      foo = MyRecord "a" 1
+        // In order to be valid the caret must be in the text "MyRecord". If the caret is inside this word (or just
+        // before the M) then the PSI structure is as follows:
+        //      ElmFunctionCallExpr  (this is the record constructor function)
```

Comment:
I like this comment. This is the sort of thing that really helps months/years from now.